### PR TITLE
WIP: Arm

### PR DESCRIPTION
**Pull request by frc6324-team-laptop**
The Arm branch contains fully documented code for the Intake and Arm subsystems, including any relevant commands they would need. The branch is currently in need of PID constant values, as they are currently default. Once they are assigned, they can be read from a text file instead of from shuffleboard to prevent accidental modifications to the constants. 
Additionally, the branch is in need of device IDs to function.
Here is the Arm Software techplan: [https://docs.google.com/document/d/1wFj2_9z4ruebLo8qN73ioKQ4loGEReYox8JBeYs9qQM/edit](url)

### CODE REVIEW

File: src/main/java/frc/robot/LEDColor.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    public void setHSV(int h, int s, int v) {
+        hue=h;
+        saturation=s;
+        value=v;
+        calcRGB();
+    }
+
+    /**
```
Comment (ckkeane) [inline src/main/java/frc/robot/LEDColor.java:69]: Nice interpolation code!

File: src/main/java/frc/robot/LEDColor.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        else if (blue > green) {
+            hue = 360 - Math.toDegrees(angle);
+        }
+    }
+
+    /**
+     * Calculates RGB from HSV
+     */
+    void calcRGB() {
+        double cMax = 255 * value;
```
Comment (ckkeane) [inline src/main/java/frc/robot/LEDColor.java]: Broadly assuming that this code is correct but its good practice to include a link to where you found this math.

File: src/main/java/frc/robot/constants/ArmConstants.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    public static final int kPivotSwitchMaxChannel = 6;
+    public static final int kEncoderSwitchMinChannel = 7;
+    public static final int kEncoderSwitchMaxChannel = 8;
+
+    public static final double kMaxHeight = 0.0; // inches
+    public static final double kArmRetractedLength = 0.0; // inches
+    public static final double kArmMaxExtensionLength = 0.0; // inches
+    public static final double kEndEffectorLength = 0.0; // inches
+    public static final double kMaxPivotAngle = 0.0; // degrees
+    public static final double kMinPivotAngle = 0.0; // degrees
```
Comment (ckkeane) [inline src/main/java/frc/robot/constants/ArmConstants.java]: Do we not have these values yet?

File: src/main/java/frc/robot/constants/ArmPresets.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ */
+public enum ArmPresets {
+    //x for left (red) hybrid goal is 25.624
+    //y for left (red) hybrid goal isn't really relevant
+    //x for all regular hybrid goals is 23.9525
+    //y for all hybrid goals is kind of 0
+    //x for all regular low shelf goals is 39.792
+    //y for all regular low shelf goals is 20.563
+    //x for all regular high shelf goals is 57.259
+    //y for all regular high shelf goals is 32.533
```
Comment (ckkeane) [inline src/main/java/frc/robot/constants/ArmPresets.java]: Can we move these into the actual enum below?

File: src/main/java/frc/robot/subsystems/StatusLED.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+                        setStripColorRGB(color1);
+                        isOn = true;
+                    }
+                }
+            }, 
+            isFinished -> {
+                timer.stop();
+            }, 
+            () -> { return false; },
+            this
```
Comment (brennan-macaig) [inline src/main/java/frc/robot/subsystems/StatusLED.java:123]: For the lines above, are we confusing `isFinished` and `end`? I think we want `end` to be `timer.stop()`, and isFinished to return false.

**Review by ckkeane — APPROVED** [review]

**Pull request merged**